python/paddle/fluid/variable_index.py

```diff
@@ -880,3 +881,538 @@ def idx_is_empty(var):
+                st = inputs['StartsTensorList']
+            else:
+                st = attrs['starts']
+            if "EndsTensorList" in inputs.keys():
+                end = inputs['EndsTensorList']
+            else:
+                end = attrs['ends']
+            if "StridesTensorList" in inputs.keys():
+                stride = inputs['StridesTensorList']
+            else:
+                stride = attrs['strides']
+            if use_strided_slice:
+                out = paddle._C_ops.strided_slice(x, axes, st, end, stride)
+                if len(decrease_axes) > 0:
+                    out = paddle._C_ops.squeeze(out, decrease_axes)
+            else:
+                out = paddle._C_ops.slice(
+                    x,
+                    axes,
+                    st,
+                    end,
+                    attrs['infer_flags'],
+                    attrs['decrease_axis'],
+                )
+        else:
+            from .framework import default_main_program
+
+            target_block = default_main_program().current_block()
+
+            slice_out_var = target_block.create_var(
+                name=unique_name.generate_with_ignorable_key(
+                    x.name + "_" + op_type
+                ),
+                dtype=x.dtype,
+            )
+            target_block.append_op(
+                type=op_type,
+                inputs=inputs,
+                outputs={'Out': [slice_out_var]},
+                attrs=attrs,
+            )
+            out = slice_out_var
+    # NOTE(zoooo0820): When all axes are decreased, the output will be 1-D
+    # with FLAGS_set_to_1d=True. In this case, one `None` should be pop out,
+    # otherwise the output shape will be not correct.
+    set_to_1d = paddle.get_flags('FLAGS_set_to_1d')['FLAGS_set_to_1d']
+    if set_to_1d and len(decrease_axes) == len(x.shape):
+        warnings.warn(
+            "Warning: In Tensor '__getitem__', if the number of scalar elements in the index is equal to the rank of the Tensor, the output should be 0-D. In order to be consistent with the behavior of previous versions, it will be processed to 1-D. But it is not correct and will be removed in release 2.6. If 1-D is still wanted, please modify the index element from scalar to slice (e.g. 'x[i]' => 'x[i:i+1]')."
+        )
+        none_axes = none_axes[1:]
+
+    if len(none_axes) > 0:
+        # Deal with cases that decrease_axes is not empty
+        # For example:
+        # # x.shape: (2,3,4)
+        # out = x[0, 0:2, None] # out.shape : (2, 1, 4)
+        for idx, axis in enumerate(none_axes):
+            l = len([i for i in decrease_axes if i < axis])
+            new_axis = axis - l
+            none_axes[idx] = new_axis
+
+        out = paddle.unsqueeze(out, axis=none_axes)
+
+    return out
+
+
+def _getitem_static(x, indices):
+    """
+    [WIP]: support __getitem__ by iteration strategy. combined indexing will be support by this.
```

Comment:
Is this WIP compliant with the current situation?

Reply:
Done, it is deleted, thanks.